ament_cmake_google_benchmark/cmake/ament_add_google_benchmark_test.cmake

```diff
@@ -0,0 +1,109 @@
+#
+# :param target: the target name which will also be used as the test name
+# :type target: string
+# :param RUNNER: the path to the test runner script (default: see
+#   ament_add_test).
+# :type RUNNER: string
+# :param TIMEOUT: the test timeout in seconds,
+#   default defined by ``ament_add_test()``
+# :type TIMEOUT: integer
+# :param WORKING_DIRECTORY: the working directory for invoking the
+#   executable in, default defined by ``ament_add_test()``
+# :type WORKING_DIRECTORY: string
+# :param SKIP_TEST: if set mark the test as being skipped
+# :type SKIP_TEST: option
+# :param ENV: list of env vars to set; listed as ``VAR=value``
+# :type ENV: list of strings
+# :param APPEND_ENV: list of env vars to append if already set, otherwise set;
+#   listed as ``VAR=value``
+# :type APPEND_ENV: list of strings
+# :param APPEND_LIBRARY_DIRS: list of library dirs to append to the appropriate
+#   OS specific env var, a la LD_LIBRARY_PATH
+# :type APPEND_LIBRARY_DIRS: list of strings
+#
+# @public
+#
+function(ament_add_google_benchmark_test target)
+  if(NOT TARGET ${target})
+    return()
+  endif()
+
+  cmake_parse_arguments(ARG
+    "SKIP_TEST"
+    "RUNNER;TIMEOUT;WORKING_DIRECTORY"
+    "APPEND_ENV;APPEND_LIBRARY_DIRS;ENV"
+    ${ARGN})
+  if(ARG_UNPARSED_ARGUMENTS)
+    message(FATAL_ERROR
+      "ament_add_google_benchmark_test() called with unused arguments: ${ARGN}")
+  endif()
+
+  set(executable "$<TARGET_FILE:${target}>")
+  set(benchmark_out "${AMENT_TEST_RESULTS_DIR}/${PROJECT_NAME}/${target}.google_benchmark.json")
+  set(common_out "${AMENT_TEST_RESULTS_DIR}/${PROJECT_NAME}/${target}.benchmark.json")
+  set(cmd
+    "${PYTHON_EXECUTABLE}" "-u" "${ament_cmake_google_benchmark_DIR}/run_and_convert.py"
+    "${benchmark_out}" "${common_out}" "--command"
+    "${executable}"
+    "--benchmark_out_format=json" "--benchmark_out=${benchmark_out}")
+  if(ARG_ENV)
+    set(ARG_ENV "ENV" ${ARG_ENV})
+  endif()
+  if(ARG_APPEND_ENV)
+    set(ARG_APPEND_ENV "APPEND_ENV" ${ARG_APPEND_ENV})
+  endif()
+  if(ARG_APPEND_LIBRARY_DIRS)
+    set(ARG_APPEND_LIBRARY_DIRS "APPEND_LIBRARY_DIRS" ${ARG_APPEND_LIBRARY_DIRS})
+  endif()
+  if(ARG_RUNNER)
+    set(ARG_RUNNER "RUNNER" ${ARG_RUNNER})
+  endif()
+  if(ARG_TIMEOUT)
+    set(ARG_TIMEOUT "TIMEOUT" ${ARG_TIMEOUT})
+  endif()
+  if(ARG_WORKING_DIRECTORY)
+    set(ARG_WORKING_DIRECTORY "WORKING_DIRECTORY" "${ARG_WORKING_DIRECTORY}")
+  endif()
+  if(ARG_SKIP_TEST)
+    set(ARG_SKIP_TEST "SKIP_TEST")
+  endif()
+
+  ament_add_test(
+    "${target}"
+    COMMAND ${cmd}
+    OUTPUT_FILE "${CMAKE_BINARY_DIR}/ament_cmake_google_benchmark/${target}.txt"
+    GENERATE_RESULT_FOR_RETURN_CODE_ZERO
+    ${ARG_SKIP_TEST}
+    ${ARG_ENV}
+    ${ARG_APPEND_ENV}
+    ${ARG_APPEND_LIBRARY_DIRS}
+    ${ARG_TIMEOUT}
+    ${ARG_WORKING_DIRECTORY}
+  )
+  set_tests_properties(
+    "${target}"
+    PROPERTIES
+    REQUIRED_FILES "${executable}"
+    LABELS "performance"
```

Comment:
Is it still intended to run these performance tests by default with all other tests?

Reply:
Ah, I forgot that we discussed that. Yeah, these should be skipped by default. I'll make that change.

Reply:
Since there isn't a clean way to use CTest labels to skip tests by default but allow you to enable them specifically, the best option I have is to use an environment variable. In order to allow the wrapper to detect the skipped test and pass that information to the `ament_cmake_test` wrapper, we need to support the `SKIP_RETURN_VALUE` CTest property properly in `ament_cmake_test`. #264 does that, and will now block this PR.

Reply:
I am not a big fan of introducing a custom environment variable which isn't really discoverable and needs to be documented.

Maybe doing what we did for `xfail` labeled tests would be better? I just think this should be a very conscious decision - maybe we need to discuss this in an upcoming meeting?

Reply:
My hesitation with using labels is that once there is an argument to skip a label, you can't un-skip any tests with that label with any additional CTest arguments. For example, if we were to skip `performance` by default, I couldn't run `google_benchmark` labeled tests without dropping the `performance` exclusion.

That said, I understand your point. Would it be appropriate to drop the environment stuff and merge this package with as it was before to unblock writing tests, and discuss the disabling mechanism next Tuesday?

Reply:
> Would it be appropriate to drop the environment stuff and merge this package as it was before to unblock writing tests, and discuss the disabling mechanism next Tuesday?

Sounds good to me.